pkg/lvm/lvm_util.go

```diff
@@ -86,7 +86,33 @@ func buildLVMCreateArgs(vol *apis.LVMVolume) []string {
 	}
 
 	LVMVolArg = append(LVMVolArg, vol.Spec.VolGroup)
+	return LVMVolArg
+}
+
+// builldThinLVMCreateArgs returns lvcreate command for thin volume
+func buildThinLVMCreateArgs(vol *apis.LVMVolume) []string {
+	var LVMVolArg []string
+
+	volume := vol.Name
+	size := vol.Spec.Capacity + "b"
+	pool := vol.Spec.VolGroup + "_thinpool"
+
+	if len(vol.Spec.Capacity) != 0 {
+		// check if thin pool exists for given volumegroup
+		if !lvThinExists(vol.Spec.VolGroup, pool) {
+			LVMVolArg = append(LVMVolArg, "-L", size)
+		}
+	}
 
+	// command to create thinpool and thin volume
+	// `lvcreate -L 1G -T lvmvg/mythinpool -V 1G -n thinvol`
+	if strings.TrimSpace(vol.Spec.ThinProvision) == "yes" {
```

Comment:
this extra check can be removed here since we are checking this already in caller